draft-ietf-tls-extended-key-update.md

```diff
@@ -797,9 +803,13 @@ This section discusses additional security and operational aspects introduced by
 
 Extended Key Update assumes a transient compromise of the current application
 traffic keys, not a persistent attacker with ongoing access to key material.
-Long-term private keys are assumed secure, and post-compromise security
-therefore remains achievable. This represents a short-term exposure of system
-memory or TLS session state after the handshake has completed.
+Long-term private keys are assumed to remain secure, as they are stored in a
+secure element, such as a Trusted Execution Environment (TEE), Hardware Security
+Module (HSM), or Trusted Platform Module (TPM). In contrast, application traffic
+keys are stored in within the rich operating system, where short-term exposure due
+to memory disclosure or transient compromise may occur. Post-compromise security
+therefore remains achievable, as a fresh key exchange during an Extended Key Update
+restores confidentiality once the compromise is resolved.
```

Comment:
I'm afraid this entire paragraph doth butter no parsnips.

- no long-term private keys are used in this entire scheme, so they cannot contribute to achieving post-compromise security?
- if the application traffic keys are compromised, then using them further to repair the connection with a key exchange whose authenticity is hooked off compromised keys, is ineffective.

Maybe this is addressing a very narrow model, where the keys are compromised but the adversary promises not to use them? If so, this text should be saying that.

Reply:
I updated the text for better clarity.